examples/VideoSpeedTest.py

```diff
@@ -157,27 +159,31 @@ def mkData():
                 dt = xp.uint16
                 loc = 4096
                 scale = 1024
-                mx = 2**16
+                mx = 2**16 - 1
             elif cacheKey[0] == 'float':
                 dt = xp.float32
                 loc = 1.0
                 scale = 0.1
                 mx = 1.0
             else:
                 raise ValueError(f"unable to handle dtype: {cacheKey[0]}")
-            
+
+            chan_shape = (height, width)
             if ui.rgbCheck.isChecked():
-                data = xp.random.normal(size=(frames,width,height,3), loc=loc, scale=scale)
-                data = pg.gaussianFilter(data, (0, 6, 6, 0))
+                frame_shape = chan_shape + (3,)
             else:
-                data = xp.random.normal(size=(frames,width,height), loc=loc, scale=scale)
-                data = pg.gaussianFilter(data, (0, 6, 6))
-            if cacheKey[0] != 'float':
-                data = xp.clip(data, 0, mx)
-            data = data.astype(dt)
-            data[:, 10, 10:50] = mx
-            data[:, 9:12, 48] = mx
-            data[:, 8:13, 47] = mx
+                frame_shape = chan_shape
+            data = xp.empty((frames,) + frame_shape, dtype=dt)
+            view = data.reshape((-1,) + chan_shape)
+            for idx in range(view.shape[0]):
+                subdata = xp.random.normal(loc=loc, scale=scale, size=chan_shape)
```

Comment:
This omits the gaussian blurring of the previous code. I'm not sure if that was in there for any important reason. I read through the history, and it has always been a part of this code, and never with explanation. It does visibly alter the image, which is maybe more indicative of e.g. microscope images. I didn't notice any fps differences either way, so it might be safest to leave it in place.

Reply:
The gaussian blurring is the function that slows down array creation the most. This becomes observable for larger matrices like 2048x2048. Large matrices are needed to make the profiling timings more measurable. It also makes changing the size on-the-fly from the GUI to be very slow, which makes it harder for experimentation.

It doesn't affect the frame rate. But it makes experimenting easier.

Reply:
That's legit. Could you leave something in as a comment for now? I'll dig a bit further to see if there's ever a good cause for smoothing things first.

Reply:
I took a look at ```pg.gaussianFilter()```. I counted at least 4 temporary copies of the input data would be in-flight within that function. So for a 4096x4096 uint8 array, the initial random float64 data would be 128MB. While passing through ```pg.gaussianFilter()```, another 512MB would be allocated.
This would be rather surprising to the user who thought his array was only 16MB.